### PR TITLE
Pricing Pattern

### LINKED ISSUE #111: Block Patterns - Pricing
**Description**

🛠️ The goal of this issue is to track the implementation of the "Pricing" pattern.
🎨 [Figma](https://www.figma.com/file/AlYr03vh4dVimwYwQkTdf6/Twenty-Twenty-Four?type=design&node-id=492%3A4275&mode=design&t=LHSPHKUrPq556l59-1)

Pricing

### PR DESCRIPTION
**Pull request by nudgeD**
Added Pricing pattern. #111

### CODE REVIEW

File: patterns/pricing.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+			<!-- wp:heading {"textAlign":"center","level":5,"style":{"spacing":{"margin":{"bottom":"var:preset|spacing|40"}}},"fontSize":"x-large"} -->
+			<h5 class="wp-block-heading has-text-align-center has-x-large-font-size" style="margin-bottom:var(--wp--preset--spacing--40)"><?php echo esc_html_x( '$0', 'Price for Free pricing level', 'twentytwentyfour' ); ?></h5>
+			<!-- /wp:heading -->
+
+			<!-- wp:paragraph {"align":"center"} -->
+			<p class="has-text-align-center"><?php echo wp_kses_post( _x( 'Access to 5 exclusive <em>Études Articles</em> per month ✓', 'Feature for pricing level', 'twentytwentyfour' ) ); ?></p>
+			<!-- /wp:paragraph -->
+
+			<!-- wp:separator {"backgroundColor":"custom-borders"} -->
```
Comment (carolinan) [inline patterns/pricing.php]: The name `custom-borders` has been changed to `base-3` and needs to be updated in the pattern:  https://github.com/WordPress/twentytwentyfour/commit/6e1a76b448ebf4fc2aa02e519076f5e8caf5d889

File: patterns/pricing.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Title: Pricing
+ * Slug: twentytwentyfour/pricing
+ * Categories: call-to-action
+ */
+
+?>
+
+<!-- wp:group {"align":"wide","style":{"spacing":{"padding":{"top":"var:preset|spacing|50","bottom":"var:preset|spacing|40","left":"var:preset|spacing|40","right":"var:preset|spacing|40"}}},"backgroundColor":"base-2","layout":{"type":"constrained"}} -->
```
Comment (carolinan) [inline patterns/pricing.php]: The group should have a border radius of 16 px:
 https://www.figma.com/file/AlYr03vh4dVimwYwQkTdf6/Twenty-Twenty-Four?node-id=492%3A4276&mode=dev

File: patterns/pricing.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+<div class="wp-block-group alignwide has-base-2-background-color has-background" style="padding-top:var(--wp--preset--spacing--50);padding-right:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40);padding-left:var(--wp--preset--spacing--40)"><!-- wp:heading {"textAlign":"center","fontSize":"x-large"} -->
+	<h2 class="wp-block-heading has-text-align-center has-x-large-font-size"><?php echo esc_html_x( 'Our Services', 'Heading for pricing pattern', 'twentytwentyfour' ); ?></h2>
+	<!-- /wp:heading -->
+
+	<!-- wp:group {"align":"wide","layout":{"type":"default"}} -->
```
Comment (carolinan) [inline patterns/pricing.php]: Why is this inner group that wraps the paragraph needed?

**Comment by carolinan** [comment]
@beafialho The light grey text color is not accessible, the contrast against the white background is too low.
(I tested using the orange, but it does not have a high enough contrast either, and it stood out too much)
What do you suggest as an alternative?

File: patterns/pricing.php (hunk 3 above)
Comment (nudgeD) [reply patterns/pricing.php]: Content width is 620px. The design has this paragraph on one line at 721px wide. Without the group, the paragraph wraps onto a second line.

File: patterns/pricing.php (hunk 2 above)
Comment (nudgeD) [reply patterns/pricing.php]: Thanks for catching that. I'll add it.

**Comment by beafialho** [comment]
@carolinan we can use `#767676` for the text, which passes contrast checks. Let's leave the lines and strokes as they are.

**Comment by richtabor** [comment]
Made a few tweaks; using contrast-2 for the lighter text + added spacers instead of margin values between elements. 

CleanShot 2023-09-05 at 14 59 37

**Review by richtabor — APPROVED** [review]

**Pull request merged**